e2e/node/e2e.test.ts

```diff
@@ -303,4 +325,355 @@ describe(`End-to-end access grant tests for environment [${environment}}]`, () =
+      const ownerDataset = await solidClient.getSolidDataset(testFileIri, {
+        fetch: resourceOwnerSession.fetch,
+      });
+
+      const requestorDataset = await getSolidDataset(testFileIri, accessGrant, {
+        fetch: requestorSession.fetch,
+      });
+
+      const ownerTtl = await solidClient.solidDatasetAsTurtle(ownerDataset);
+      const requestorTtl = await solidClient.solidDatasetAsTurtle(
+        requestorDataset
+      );
+
+      expect(ownerTtl).toBe(requestorTtl);
+    });
+
+    it("can use the saveSolidDatasetAt API for an existing dataset", async () => {
+      await expect(
+        isValidAccessGrant(accessGrant, {
+          fetch: resourceOwnerSession.fetch,
+          // FIXME: Currently looking up JSON-LD doesn't work in jest tests.
+          // It is an issue documented in the VC library e2e test, and in a ticket
+          // to be fixed.
+          verificationEndpoint: VERIFY_ENDPOINT,
+        })
+      ).resolves.toMatchObject({ errors: [] });
+
+      // Here we request the dataset as the resource owner, but in real-world
+      // applications, you'd need to use an Access Grant to request the dataset
+      // as the requestor, this is just to limit how much of the Access Grants
+      // library we're testing in a single test case:
+      const dataset = await solidClient.getSolidDataset(testFileIri, {
+        fetch: resourceOwnerSession.fetch,
+      });
+
+      // Create a thing and add it to the dataset:
+      let newThing = solidClient.createThing({
+        name: "e2e-test-thing",
+      });
+      newThing = solidClient.setBoolean(
+        newThing,
+        "https://arbitrary.vocab/regular-predicate",
+        true
+      );
+      const datasetUpdate = solidClient.setThing(dataset, newThing);
+
+      // Try to update the dataset as a requestor of the Access Grant:
+      const updatedDataset = await saveSolidDatasetAt(
+        testFileIri,
+        datasetUpdate,
+        accessGrant,
+        {
+          fetch: requestorSession.fetch,
+        }
+      );
+
+      // Fetch it back as the owner to prove the dataset was actually updated:
+      const updatedDatasetAsOwner = await solidClient.getSolidDataset(
+        testFileIri,
+        {
+          fetch: resourceOwnerSession.fetch,
+        }
+      );
+
+      // Serialize each to turtle:
+      const updatedDatasetTtl = await solidClient.solidDatasetAsTurtle(
+        updatedDataset
+      );
+      const existingDatasetAsOwnerTtl = await solidClient.solidDatasetAsTurtle(
+        dataset
+      );
+      const updatedDatasetAsOwnerTtl = await solidClient.solidDatasetAsTurtle(
+        updatedDatasetAsOwner
+      );
+
+      // Assert that the dataset changed:
+      expect(updatedDatasetTtl).not.toBe(existingDatasetAsOwnerTtl);
+      expect(updatedDatasetTtl).toBe(updatedDatasetAsOwnerTtl);
+    });
+
+    it.skip("can use the saveSolidDatasetAt API for a new dataset", async () => {
```

Comment:
@NSeydoux hey, I wasn't able to get this working as an e2e test, but we've had it as an API for a while. I'm not sure if this would ever work as expected, given the way these APIs are written.